### DISCUSSION TITLE
/e2e-testing-microsoft-teams-solutions-cypress/

### DISCUSSION THREAD

**Discussion by comment-migrator** [estruyf/eliostruyf-comments#510] [Comments]
End-to-End testing your Microsoft Teams solutions with Cypress
        
[https://www.eliostruyf.com/e2e-testing-microsoft-teams-solutions-cypress/](https://www.eliostruyf.com/e2e-testing-microsoft-teams-solutions-cypress/)

**Comment by comment-migrator** [comment]
```header
Originally posted by Ross Johnson on 2020-07-30 01:15
```

Thanks for sharing this! For organizations that haven't settled on a test tool, are there any others they might consider before selecting Cypress? Do you think any would be more capable of testing the alternative user interactions we see in solutions spanning full clients of Teams, Tasks, Outlook etc?

**Reply by comment-migrator** [reply]
```header
Originally posted on 2020-07-31 16:54
```

I haven't tested tools for full client testing. For sure they exists, but our products are web based. So we only need to test in a browser.

Other tools you could consider for web UI testing are Selenium, Puppeteer or Playwright in combination with a testing framework like Jest (example: [https://www.eliostruyf.com/...](https://www.eliostruyf.com/utilize-playwright-jest-cross-browser-e2e-test-solutions/) "https://www.eliostruyf.com/utilize-playwright-jest-cross-browser-e2e-test-solutions/)").

I'm not a big fan of Selenium, as it's a bit out dated, but still works. Playwright is a very good option as well, but I like the UI which Cypress provides to quickly run individual or complete tests and see the current outcome. This makes it easy for testers and developers for checking if a specific functionality works. When running during CI/CD, you could rely on its screenshot and video recorder.

**Comment by comment-migrator** [comment]
```header
Originally posted by Rob Brennan on 2020-09-02 17:48
```

This looks promising! I'm trying to find a way to get this to work against an app using react-adal and Azure AD logins. No love so far, though. Any suggestions? Basically just trying to get it to verify the user can view a url, sign in with that new iframe that is spun off for MS login, and then verify they are authenticated against that same url 🤷‍♂️

**Reply by comment-migrator** [reply]
```header
Originally posted on 2020-09-15 08:30
```

I think you could use the same approach as I did for SharePoint with Azure AD and ADAL. Here is the article about it: [https://www.eliostruyf.com/...](https://www.eliostruyf.com/cypress-test-sharepoint-solution-azure-ad-secured-api/ "https://www.eliostruyf.com/cypress-test-sharepoint-solution-azure-ad-secured-api/")

Basically, you do the auth process and token retrieval in Cypress and set the token in your session storage.

**Comment by comment-migrator** [comment]
```header
Originally posted by Ken Ho on 2021-05-14 23:02
```

Not sure if Cypress still works with [https://teams.microsoft.com/](https://teams.microsoft.com/ "https://teams.microsoft.com/").

I am able to set tokens, but teams won't load in Cypress iframe, and no error messages in console.log. But it can load in a new tab in Cypress(not within the iframe).

Video:

[https://www.youtube.com/wat...](https://www.youtube.com/watch?v=F0rhz24gU3s "https://www.youtube.com/watch?v=F0rhz24gU3s")

**Reply by comment-migrator** [reply]
```header
Originally posted on 2021-09-10 08:39
```

Sorry, for some reason your comment was marked as spam.

Did you already find a solution to your problem?

**Comment by comment-migrator** [comment]
```header
Originally posted by Lawrence Cabal on 2021-09-14 21:29
```

Great article. Helped me a lot. Have you tried with multiple users? What I want to happen is load cookies and localStorage for user 1. Go to conversation and say hi. Then, clear cookies and localstorage. Load cookies and localstorage of user 2. Go to conversaation and say hi again. This gives me an error.

any suggestion?

**Reply by comment-migrator** [reply]
```header
Originally posted by Lawrence Cabal on 2021-09-16 05:08
```

I got it to work. It wasn't working because when i was loading the cookies i had Cypress.Cookies.preserveOnce. Once I removed that I had no problem swtching from user 1 to user 2.

**Comment by alexsandro134** [comment]
Thanks for your article, it gives me the idea about the root cause of issue I had faced. Just want to update the solution at the moment: in cypress experiments, it has been release an option `experimentalModifyObstructiveThirdPartyCode`. Enable this option will handle the issue with microsoft teams.

**Reply by estruyf** [reply]
Thank you, @alexsandro134 for sharing this information!
